lib/bloc/blocs/auth_provider.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/material.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:structure_flutter/core/common/enums/authenticate.dart';
+import 'file:///G:/Project/jobchat/Structure_Flutter/lib/widgets/snackbar_service.dart';
+import 'package:structure_flutter/core/extension/navigation_extension.dart';
+import 'package:structure_flutter/repositories/account_repository.dart';
+
+class AuthProvider extends ChangeNotifier {
+  User user;
+  AuthStatus status;
+
+  FirebaseAuth auth;
+
+  static AuthProvider instance = AuthProvider();
+
+  AuthProvider() {
+    auth = FirebaseAuth.instance;
+    _checkCurrentUserIsAuthenticated();
+  }
+
+  void _autoLogin() async {
+    if (user != null) {
+      await AccountRepositoryImpl.instance.updateUserLastSeenTime(user.uid);
+      return NavigationExtension.instance.navigateToReplacement("home");
```

Comment:
The purpose we use ```Provider``` to separate ui and business logic. So dont write ui logic inside business logic component.